facia/app/services/PressedPageService.scala

```diff
@@ -0,0 +1,54 @@
+package services
+
+import common.{FaciaPressMetrics, Logging, StopWatch}
+import concurrent.FutureSemaphore
+import model.PressedPage
+import play.api.libs.json.{JsError, JsSuccess, JsValue, Json}
+import play.api.libs.ws.WSClient
+
+import scala.concurrent.{ExecutionContext, Future}
+
+class PressedPageService(wsClient: WSClient) extends Logging {
+  val secureS3Request = new SecureS3Request(wsClient)
+  val futureSemaphore = new FutureSemaphore(6)
+
+  def get(path: String)(implicit executionContext: ExecutionContext): Future[Option[PressedPage]] = errorLoggingF(s"PressedPageService.get $path") {
+    futureSemaphore.execute(getPressedPage(path)).map(_.get)
```

Comment:
calling `_.get` on the `Try` since I don't want to introduce error handling here for simplicity. Happy to simply log and return an `internal server error`

Reply:
ok but will it log the original error or just the error from the Failure.get?

Reply:
Failure.get is the original error

Reply:
I did not know that Failure.get throws the exception contained! but that makes good sense.

ageed with your second point about the stack trace.